source/MaterialXRender/Handlers/ImageHandler.h

```diff
@@ -56,12 +64,33 @@ class ImageDesc
     ImageType imageType = IMAGETYPE_2D;
     /// Hardware target dependent resource identifier. May be undefined.
     unsigned int resourceId = 0;
+    /// Deallocator to free resource buffer memory. If not defined then malloc() is
+    /// assumed to have been used to allocate the buffer and corresponding free() is
+    /// used to deallocate.
+    ImageBufferDeallocator resourceBufferDeallocator = nullptr;
```

Comment:
New deallocator function ptr.

Reply:
Instead of setting nullptr, maybe you can set `free` by default here? And in freeResourceBuffer you don't have to handle the 'free' case explicitly.